get_smbclient_test.go

```diff
@@ -0,0 +1,327 @@
+			"smb://",
+			"",
+			true,
+		},
+		{
+			"no filepath provided",
+			"smb://samba",
+			"",
+			true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dst := testing_helper.TempDir(t)
+			defer os.RemoveAll(dst)
+
+			url, err := urlhelper.Parse(tt.rawURL)
+			if err != nil {
+				t.Fatalf("err: %s", err.Error())
+			}
+			req := &Request{
+				Dst: filepath.Join(dst, tt.file),
+				u:   url,
+			}
+
+			g := new(SmbClientGetter)
+			err = g.GetFile(context.Background(), req)
+
+			fail := err != nil
+			if tt.fail != fail {
+				if fail {
+					t.Fatalf("err: unexpected error %s", err.Error())
+				}
+				t.Fatalf("err: expecting to fail but it did not")
+			}
+
+			if !tt.fail {
+				// Verify if the file was successfully downloaded
+				// and exists at the destination folder
+				testing_helper.AssertContents(t, filepath.Join(dst, tt.file), "Hello\n")
+			}
+		})
+	}
+}
+
+func TestSmb_GetterMode(t *testing.T) {
+	smbTestsPreCheck(t)
+
+	tests := []struct {
+		name         string
+		rawURL       string
+		expectedMode Mode
+		fail         bool
+	}{
+		{
+			"smbclient modefile for existing file in authenticated private share",
+			"smb://user:password@samba/private/file.txt",
+			ModeFile,
+			false,
+		},
+		{
+			"smbclient modedir for existing directory in authenticated private share",
+			"smb://user:password@samba/private/subdir",
+			ModeDir,
+			false,
+		},
+		{
+			"mode fail for non existent directory in authenticated private share",
+			"smb://user:password@samba/private/invaliddir",
+			0,
+			true,
+		},
+		{
+			"mode fail for non existent file in authenticated private share",
+			"smb://user:password@samba/private/invalidfile.txt",
+			0,
+			true,
+		},
+		{
+			"smbclient modefile for existing file in public share",
+			"smb://samba/public/file.txt",
+			ModeFile,
+			false,
+		},
+		{
+			"smbclient modedir for existing directory in public share",
+			"smb://samba/public/subdir",
+			ModeDir,
+			false,
+		},
+		{
+			"mode fail for non existent directory in public share",
+			"smb://samba/public/invaliddir",
+			0,
+			true,
+		},
+		{
+			"mode fail for non existent file in public share",
+			"smb://samba/public/invalidfile.txt",
+			0,
+			true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			url, err := urlhelper.Parse(tt.rawURL)
+			if err != nil {
+				t.Fatalf("err: %s", err.Error())
+			}
+
+			g := new(SmbClientGetter)
+			mode, err := g.Mode(context.Background(), url)
+
+			fail := err != nil
+			if tt.fail != fail {
+				if fail {
+					t.Fatalf("err: unexpected error %s", err.Error())
+				}
+				t.Fatalf("err: expecting to fail but it did not")
+			}
+
+			if mode != tt.expectedMode {
+				t.Fatalf("err: expeting mode %d, actual mode %d", tt.expectedMode, mode)
+			}
+		})
+	}
+}
+
+func smbTestsPreCheck(t *testing.T) {
+	r := os.Getenv("ACC_SMB_TEST")
+	if r != "1" {
+		t.Skip("smb getter tests won't run. ACC_SMB_TEST not set")
+	}
+}
```

Comment:
what do you think about putting this func at the top of the file with a comment so that users understands that all tests in this file can be skipped here.

Also, super good job on the smb server in circle.